### ISSUE TITLE
Item dupe by restarting missions

### ISSUE THREAD

**Issue by geist22** [mefistotelis/swars#16]
_(I will be creating issues here for the things I've found, Most are part of the original game, but I will try and doublecheck them in the port beforehand as I go, hope it won't be too spammy)_

Known issue from the original game.

If you pick up items with multiple uses (like High Explosives), and then restart the mission, you keep the item but it doesn't show up on the weapon selection menu until you add/remove one from the stack. So you can pick up a High Explosive, restart the mission, pick it up again and repeat until all agents have 4 each. This way you can softlock the space station mission as you can neither pick up nor drop any explosives anymore.

I've tested this with High Explosives and Knockout Gas, it should probably work with all items of that type (Psycho Gas and Nuclear Grenade too, not counting Ion Mine because they cannot be picked up as far as I know).

**Comment by mefistotelis** [comment]
Most likely fixed by [swars: Always set amount of consumable weapons](https://github.com/mefistotelis/swars/commit/979c293350ebed3e62149fc65ca716e9c695e607) and  [Remade do_weapon_quantities_proper1()](https://github.com/mefistotelis/swars/commit/080ef3d0074d846191dfb07a21bd3eeaba52980d).

Needs testing though.

**Comment by geist22** [comment]
I did a quick test with High Explosives in Eurocorp Mission 1 (the second mission, as the first is mission 0), and it did fix agents keeping invisible items through mission restarts for the few times I've tried it.

Does this also affect NPC inventory (e.g. enemies not dropping an item if they've used it)? If so that should be toggleable as this might have gameplay changing implications. Though that doesn't seem to be the case, the female Unguided in the lower left area drops only 1 Explosive when killed but will usually place 4 during her scripted run (sometimes she skips placing the first one for some reason, I don't know why).
